packages/compiler/src/render3/r3_template_transform.ts

```diff
@@ -151,10 +151,14 @@ class HtmlAstToIvyAst implements html.Visitor {
         const templateKey = normalizedName.substring(TEMPLATE_ATTR_PREFIX.length);
 
         const parsedVariables: ParsedVariable[] = [];
-        const absoluteOffset = attribute.valueSpan ? attribute.valueSpan.start.offset :
-                                                     attribute.sourceSpan.start.offset;
+        const absoluteValueOffset = attribute.valueSpan ?
+            attribute.valueSpan.start.offset :
+            // If there is no value span the attribute does not have a value, like `attr` in
+            //`<div attr></div>`. In this case, point to one beyond the attribute name.
```

Comment:
Shouldn't this be more like this?

```suggestion
            //`<div attr></div>`. In this case, point to the end of the attribute name.
```

Reply:
I mean to say that the offset of the attribute value is one character past the last character of the attribute name. I have clarified that comment. It shouldn't point to the last character of the attribute name.

Reply:
Yeah, but I don't see the one character offset in the code. Adding the attribute name's length ends up at the end of the attribute name, not one character beyond it? Am I missing something here?

Reply:
I believe it does:

```
word
^^^^^
01234
```

so adding `0 + len(word) = 4` which is one beyond the last character of the word. Let me know if I'm misinterpreting something.